library/systemd/src/lib/yast2/systemd_socket_finder.rb

```diff
@@ -0,0 +1,100 @@
+# encoding: utf-8
+
+# Copyright (c) [2018] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require "yast"
+require "yast2/execute"
+
+Yast.import "Systemctl"
+Yast.import "Stage"
+
+module Yast2
+  # This class is responsible for finding out which sockets trigger a given service.
+  #
+  # When systemd is working properly, the relationship between services and sockets is cached in
+  # order to reduce the amount of calls to `systemctl show`. However, during the installation, where
+  # systemd is not fully operational, this class just tries to find a socket named after the
+  # service.
+  #
+  # @example Finding a socket
+  #   finder = Yast2::SystemdSocketFinder.new
+  #   finder.for_service("cups").class # => Yast::SystemdSocketClass::Socket
+  class SystemdSocketFinder
+    # Returns the socket for a given service
+    #
+    # @return [Yast2::SystemdSocketClass::Socket,nil]
+    def for_service(service_name)
+      socket_name = socket_name_for(service_name)
+      return nil unless socket_name
+      Yast::SystemdSocket.find(socket_name)
+    end
+
+  private
+
+    # Return the socket's name for a given service
+    #
+    # @note On 1st stage it returns just the same name.
+    #
+    # @return [String,nil]
+    def socket_name_for(service_name)
+      return service_name if Yast::Stage.initial
+      sockets_map[service_name]
+    end
+
+    # Builds a map between services and sockets
+    #
+    # @return [Hash<String,String>] Sockets indexed by the name of the service they trigger
+    def sockets_map
```

Comment:
What happens when a service is triggered by several sockets? I guess it takes the last socket.

Reply:
Yes, that's it. I would document that behaviour, because it is something I would like to discuss for the future.